### PR TITLE
Don't clip when the clip path is empty (issue #12306)

### LINKED ISSUE #12306: All empty pages in pdf file
Configuration:
- Web browser and its version: Chrome
- Operating system and its version: Windows 10
- PDF.js version: latest
- Is a browser extension: no

Checked on demo website:
https://mozilla.github.io/pdf.js/web/viewer.html

Steps to reproduce the problem:
1. Upload attached file.
2. All pages are empty.
[document13-08-2020-122237.pdf](https://github.com/mozilla/pdf.js/files/5150027/document13-08-2020-122237.pdf)

Expected:
Acrobat reader can open this document.
scevkRtfnQ

### PR DESCRIPTION
**Pull request by calixteman**

### CODE REVIEW

File: test/test_manifest.json [hunk 1]
```diff
@@ -6362,5 +6362,13 @@
       "link": true,
       "lastPage": 1,
       "type": "eq"
+   },
+   {  "id": "issue12036",
+      "file": "pdfs/issue12036.pdf",
```
Comment (Snuffleupagus) [inline test/test_manifest.json:6367]: There's a typo in the issue number on both of these lines :-)

**Comment by Snuffleupagus** [comment]
Unfortunately there appears to be a number of regressions, see an *incomplete* list below:
 - issue13447-eq-page1
 - bug1669099-page1
 - There's also movement in what appears to be most, if not all, Type3-related tests

**Comment by calixteman** [comment]
> Unfortunately there appears to be a number of regressions, see an _incomplete_ list below:
> 
>     * issue13447-eq-page1
> 
>     * bug1669099-page1
> 
>     * There's also movement in what appears to be most, if not all, Type3-related tests

Likely related to https://github.com/mozilla/pdf.js/blob/379125c37ff58cfb86e3bbd2eca5760fb2fcb061/src/display/canvas.js#L2823
I changed it to `this.ctx.clip()` which should be better.

**Comment by Snuffleupagus** [comment]
> Likely related to
> 
> https://github.com/mozilla/pdf.js/blob/379125c37ff58cfb86e3bbd2eca5760fb2fcb061/src/display/canvas.js#L2823
> 
> I changed it to `this.ctx.clip()` which should be better.

That really cannot explain all of the movement in Type3-tests though, since the code-path in question only matter for annotations...

**Comment by Snuffleupagus** [comment]
I completely missed that there was a regressions in the `tiling-pattern-large-steps-page1` test-case (look at its *right* edge).

@calixteman Can you please take a look into that, and fix it in a follow-up?

**Comment by calixteman** [comment]
Yep, it's likely because of https://github.com/mozilla/pdf.js/blob/5bbed400f2e82422c5bd7cc507b2c14cc1dc16f1/src/display/pattern_helper.js#L622 (it's just a guess).

**Pull request merged**